sympy/integrals/tests/test_integrals.py

```diff
@@ -1170,3 +1170,7 @@ def test_issue_4950():
 
 def test_issue_4968():
     assert integrate(sin(log(x**2))) == x*sin(2*log(x))/5 - 2*x*cos(2*log(x))/5
+
+def test_singularities():
+    assert integrate(1/x**2,(x,-oo,oo)) == oo
+    assert integrate(1/(x - 1)**2,(x,-2,2)) == oo
```

Comment:
I think `integrate()` expects `a <= b`. This is the following error on master branch : 
```
>>> integrate(1/x**2,(x,oo,-oo))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/integrals/integrals.py", line 1295, in integrate
    risch=risch, manual=manual)
  File "sympy/integrals/integrals.py", line 466, in doit
    ret = try_meijerg(function, xab)
  File "sympy/integrals/integrals.py", line 443, in try_meijerg
    res = meijerint_definite(function, x, a, b)
  File "sympy/integrals/meijerint.py", line 1803, in meijerint_definite
    return -meijerint_definite(f, x, b, oo)
TypeError: bad operand type for unary -: 'NoneType'
```
I have performed the other requested changes.